pam/internal/adapter/authentication.go

```diff
@@ -45,8 +51,9 @@ func sendIsAuthenticated(ctx context.Context, client authd.PAMClient, sessionID
 		}
 
 		return isAuthenticatedResultReceived{
-			access: res.Access,
-			msg:    res.Msg,
+			access:    res.Access,
+			msg:       res.Msg,
+			challenge: challenge,
```

Comment:
I told you before that I didn't agree with this logic for validating the password and you still added it. 

There's a function in the `nativemodel.go` file that is supposed to handle new password conversations (and therefore have access to the new password input by the user), so the challenge check should be there. The broker returns the access level and a message (that's either an error or the user data), so there's no reason for us to store the challenge (that might not even exist) in this type as well.

Reply:
Mh, it's something I had refactored days ago, so I thought you had seen it already, but this commit can be probably dropped anyways here while I will need something similar in GDM, since we need to communicate through these messages.

Reply:
Removed.